src/Results.Immutable/Option.cs

```diff
@@ -22,6 +24,36 @@
     ///     A new instance of <see cref="Immutable.Option{T}" /> without value.
     /// </returns>
     public static Option<T> None<T>() => new();
+
+    /// <summary>
+    /// Converts a nullable struct type to an <see cref="Option{T}" />,
+    /// if the value is null, the <see cref="Option{T}" /> is None.
+    /// </summary>
+    /// <typeparam name="T">Type of the value.</typeparam>
+    public static Option<T> FromNullable<T>(T? value) where T : struct =>
+            value.HasValue ? Some(value.Value) : None<T>();
+
+    /// <summary>
+    /// Converts a nullable reference type to an <see cref="Option{T}" />,
+    /// if the value is null, the <see cref="Option{T}" /> is None.
+    /// </summary>
+    /// <typeparam name="T">Type of the value.</typeparam>
+    public static Option<T> FromNullable<T>(T? value) where T : class =>
+          value is not null ? Some(value) : None<T>();
+
+
+    /// <summary>
+    /// Converts an <see cref="IEnumerable{T}" /> into an <see cref="Option{T}" />,
+    /// it takes the first element of the <see cref="IEnumerable{T}" /> and the rest are discarded.
+    /// If the <see cref="IEnumerable{T}" /> is empty, the <see cref="Option{T}" /> is None.
+    /// </summary>
+    /// <param name="enumerable">An <see cref="IEnumerable{T}"/> to convert into an <see cref="Option{T}"/>.</param>
+    /// <typeparam name="T">Type of the value.</typeparam>
+    public static Option<T> FromEnumerable<T>(IEnumerable<T> enumerable)
+    {
+        var enumerator = enumerable.GetEnumerator();
+        return enumerator.MoveNext() ? Some(enumerator.Current) : None<T>();
+    }
```

Comment:
`.Any()` without passing a predicate does not enumerate in most cases, so it might be worth it to implement this like so:
```suggestion
    public static Option<T> FromEnumerable<T>(IEnumerable<T> enumerable) =>
        enumerable.Any() ? Some(enumerable.First()) : None<T>();
```

Reply:
I don't agree to use linq methods when all we need is the enumerator. As you say, your suggestion may lead to multiple enumerations. And in my opinion, it brings complexity (calling 2 methods) for no good reason.

Reply:
It calls the same exact amount of methods (`GetEnumerator` + `MoveNext`) so that point is moot. But I agree that in not all of the cases suggested solution will enumerate once.